File: aesi-eclipse/bundle/src/main/java/com/virtlink/aesi/syntaxhighlighting/IToken.java
package com.virtlink.aesi.syntaxhighlighting;

import com.virtlink.aesi.Span;
import com.virtlink.aesi.Span2;

/**
 * A colorizer token.
 */
public interface IToken {
	/**
	 * Gets the location of the token relative to the start of the text.
	 * 
	 * @return The start and end of the token.
	 */
	Span getLocation();
	
	/**
	 * Gets the scope of the token.
	 * 
	 * The token's scope determines how the token is colored.
	 * 
	 * @return The name of the scope.
	 */
	String getScope();
}

File: aesi/src/main/kotlin/com/virtlink/editorservices/syntaxcoloring/IToken.kt
package com.virtlink.editorservices.syntaxcoloring

import com.virtlink.editorservices.Span
import java.io.Serializable

/**
 * A coloring token.
 */
interface IToken : Serializable {

    /**
     * Gets the span in the document that is being colored.
     */
    val location: Span

    /**
     * Gets the name assigned to the token.
     *
     * The name is used to find an appropriate color for the token.
     */
    val name: String

    /**
     * Gets the attributes that apply to the token.
     *
     * This is an unordered list of names, each of which
     * indicates an attribute of the token.
     * For example "invalid.deprecated" to
     * indicate the token should display
     * as a deprecated element.
     */
    val attributes: List<String>

}